nil/services/synccommittee/prover/tracer/state_db.go

```diff
@@ -489,17 +563,18 @@ func (tsdb *TracerStateDB) SetCode(addr types.Address, code []byte) error {
 	return nil
 }
 
-func (tsdb *TracerStateDB) AddRefund(uint64) {
-	panic("not implemented")
+func (tsdb *TracerStateDB) AddRefund(gas uint64) {
+	tsdb.refund += gas
 }
 
-func (tsdb *TracerStateDB) SubRefund(uint64) {
-	panic("not implemented")
+func (tsdb *TracerStateDB) SubRefund(gas uint64) {
+	check.PanicIff(gas > tsdb.refund, "Refund counter below zero (gas: %d > refund: %d)", gas, tsdb.refund)
+	tsdb.refund -= gas
 }
 
 // GetRefund returns the current value of the refund counter.
 func (tsdb *TracerStateDB) GetRefund() uint64 {
-	panic("not implemented")
+	panic("unreachable") // Not used in ExecutionState for now
```

Comment:
if PR still in progress, let move to draft

Reply:
I meant, it is not used in `ExecuitonState` neither